platform-api/src/main/java/io/datacater/core/deployment/DeploymentEndpoint.java

```diff
@@ -77,24 +75,24 @@ public Uni<ListMeta> getDeployments() {
   }
 
   @POST
-  @Path("{pipelineUuid}")
-  public Uni<String> createDeployment(@PathParam("pipelineUuid") UUID pipelineId) {
+  public Uni<UUID> createDeployment(@QueryParam("pipelineUuid") UUID pipelineId) {
     return apply(pipelineId);
   }
 
   @DELETE
-  @Path("{deploymentName}")
-  public Uni<Response> deleteDeployment(@PathParam("deploymentName") String deploymentName) {
-    return Uni.createFrom().item(deleteK8Deployment(deploymentName));
+  @Path("{uuid}")
+  public Uni<Response> deleteDeployment(@PathParam("uuid") UUID deploymentId) {
+    return Uni.createFrom().item(deleteK8Deployment(deploymentId));
   }
 
   @PUT
-  @Path("{pipelineUuid}")
-  public Uni<String> updateDeployment(@PathParam("pipelineUuid") UUID pipelineId) {
+  @Path("{uuid}")
+  public Uni<UUID> updateDeployment(
+      @PathParam("uuid") UUID deploymentUuid, @QueryParam("pipelineUuid") UUID pipelineId) {
```

Comment:
Since `pipelineUuid` is part of the deployment spec, could we also accept a deployment spec as request body here, assuming that we do this as well for the "update" endpoints of other resources?

Reply:
We removed the deploymentSpec since we aren't persisting it and it only contained pipelineID and stream-in/stream-out which are already referenced in the PipelineObject. 

I'll re-add it to stay consistant with the other endpoints and since we will need it again at some point for K8 resource config and to change the deployments stream-in/out